doc/man3/EVP_DigestSignInit.pod

```diff
@@ -29,6 +29,13 @@ EVP_PKEY_CTX of the signing operation will be written to B<*pctx>: this can
 be used to set alternative signing options. The digest B<type> may be NULL if
 the signing algorithm supports it.
 
+In some rare cases (including SM2 signatures) it is necessary to reinitialize
+the digest computation after setting some parameters on the B<pctx>. This
+is done by calling EVP_DigestSignInit() twice. The first time, B<pctx>
```

Comment:
I'm no expert in English, but I find "twice" confusing. I give an impression that I actually have to call it two times in a row prior I can do anything.

Reply:
I can see the "twice" in test and ... it's terrible...

Reply:
I agree it is, uh, well terrible is a pretty good word for it tbh. It was suggested by mattcaswell in #6534 and I couldn't come up with an alternative that would be acceptable.

Reply:
Yes. It's quite nasty. The other alternative I suggested in #6534 was to create a new version of `EVP_DigestSignInit`. But I'm not especially keen on that idea either (it seems like a "sledgehammer to crack a nut"). IMO the original approach in #6534 was worse, so I'm out of ideas. This seemed like the least bad option.

Reply:
@dot-asm would it make it better if we reword this to clarify? Such as:

"
The SM2 algorithm accepts a personalization string, if you do not need to customize the personalization string, you can use `EVP_DigestSignInit()` as expected (just call once and use it). 

However, if you need to customize the personalization string, such as processing of SM2 signatures, it is necessary to reinitialize the digest computation after setting some parameters on the `B<pctx>` by calling `EVP_DigestSignInit()` *twice*. The rationale for this is described in https://github.com/openssl/openssl/issues/6534.

* First, call EVP_DigestSignInit() to get a handle on the `EVP_PKEY_CTX`. The first time, `B<pctx>` should be non `NULL`.
* Then, set the necessary options on `B<pctx>`, for instance, using `EVP_PKEY_CTX_ctrl_str()`.
* Afterwards, `EVP_SignVerifyInit()` is called again, using the same verification context `B<ctx>`.

"

Reply:
> the least bad option.

Which in turn is like worst excuse. As already said [much] earlier, if problem can't be solved in reasonable manner, we would be better off not solving it in minor release. Because introduction of kludges obliges us to maintain them. And they are likely to be dead end, in sense that in major release problem is likely to be solved differently. It's really loose-loose...

However. As far as I can tell, if EVP_DigestVerifyUpdate involved pmeth in performing updates, as opposite to be just an alias to EVP_DigestUpdate, single call would suffice. Right? I mean pmeth could catch first update and inject additional [personalization] data. It should be noted that EVP_MD_CTX has "virtual" update function. ("Virtual" is reference to C++-style virtual methods). This mean that pmeth can get involved in performing updates by overriding update function in EVP_MD_CTX. In this case it can even restore original function after first call, so that it wouldn't have to be involved past first invocation...

On related note I'd like to argue against naming new ctrl as sm2_uid. In sense that I'd prefer more generic name that could be reusable. For example why not personalization?